server/channels/store/sqlstore/property_field_store.go

```diff
@@ -0,0 +1,339 @@
+		return nil, errors.Wrap(err, "property_field_get_tosql")
+	}
+
+	rows, err := s.GetReplica().Query(queryString, args...)
+	if err != nil {
+		return nil, errors.Wrap(err, "property_field_get_select")
+	}
+	defer rows.Close()
+
+	field, err := propertyFieldFromRows(rows)
+	if err != nil {
+		return nil, errors.Wrap(err, "property_field_get_propertyfieldfromrows")
+	}
+
+	return field, nil
+}
+
+func (s *SqlPropertyFieldStore) GetMany(ids []string) ([]*model.PropertyField, error) {
+	queryString, args, err := s.getQueryBuilder().
+		Select(propertyFieldColumns...).
+		From("PropertyFields").
+		Where(sq.Eq{"id": ids}).
+		ToSql()
+	if err != nil {
+		return nil, errors.Wrap(err, "property_field_get_many_tosql")
+	}
+
+	rows, err := s.GetReplica().Query(queryString, args...)
+	if err != nil {
+		return nil, errors.Wrap(err, "property_field_get_many_query")
+	}
+	defer rows.Close()
+
+	fields, err := propertyFieldsFromRows(rows)
+	if err != nil {
+		return nil, errors.Wrap(err, "property_field_get_many_propertyfieldfromrows")
+	}
+
+	if len(fields) < len(ids) {
+		return nil, errors.New("property_field_get_many_missmatch_results")
+	}
+
+	return fields, nil
+}
+
+func (s *SqlPropertyFieldStore) SearchPropertyFields(opts model.PropertyFieldSearchOpts) ([]*model.PropertyField, error) {
+	if opts.Page < 0 {
+		return nil, errors.New("page must be positive integer")
+	}
+
+	if opts.PerPage < 0 {
+		return nil, errors.New("per page must be positive integer")
+	}
+
+	query := s.getQueryBuilder().
+		Select(propertyFieldColumns...).
+		From("PropertyFields").
+		OrderBy("CreateAt ASC").
+		Offset(uint64(opts.Page * opts.PerPage)).
+		Limit(uint64(opts.PerPage))
+
+	if !opts.IncludeDeleted {
+		query = query.Where(sq.Eq{"DeleteAt": 0})
+	}
+
+	if opts.GroupID != "" {
+		query = query.Where(sq.Eq{"GroupID": opts.GroupID})
+	}
+
+	if opts.TargetType != "" {
+		query = query.Where(sq.Eq{"TargetType": opts.TargetType})
+	}
+
+	if opts.TargetID != "" {
+		query = query.Where(sq.Eq{"TargetID": opts.TargetID})
+	}
+
+	queryString, args, err := query.ToSql()
+	if err != nil {
+		return nil, errors.Wrap(err, "property_field_search_tosql")
+	}
+
+	rows, err := s.GetReplica().Query(queryString, args...)
+	if err != nil {
+		return nil, errors.Wrap(err, "property_field_search_query")
+	}
+	defer rows.Close()
+
+	fields, err := propertyFieldsFromRows(rows)
+	if err != nil {
+		return nil, errors.Wrap(err, "property_field_search_propertyfieldfromrows")
+	}
+
+	return fields, nil
+}
+
+func (s *SqlPropertyFieldStore) Update(fields []*model.PropertyField) (_ []*model.PropertyField, err error) {
+	if len(fields) == 0 {
+		return nil, nil
+	}
+
+	transaction, err := s.GetMaster().Beginx()
+	if err != nil {
+		return nil, errors.Wrap(err, "property_field_update_begin_transaction")
+	}
+	defer finalizeTransactionX(transaction, &err)
+
+	updateTime := model.GetMillis()
+	for _, field := range fields {
+		field.UpdateAt = updateTime
+
+		if err := field.IsValid(); err != nil {
```

Comment:
I'm wondering if using `err :=` will shadow the `err` declared in the function definition (`(_ []*model.PropertyField, err error)`) and ultimately impact what `defer finalizeTransactionX(transaction, &err)` does

Reply:
The returned value is the explicit one instead of the `err`, but happy to make this more explicit to avoid confusion

Reply:
PTAL b3f4648131